doc/langdef.md

```diff
@@ -441,6 +437,38 @@ Also note that all of these conversions are dynamic at runtime, so CEL's static
 type analysis cannot avoid the possibility of type-related errors in expressions
 using these dynamic values.
 
+## JSON Data Conversion
+
+CEL can also work with JSON data. Since there is a natural correspondence of
+most CEL data with protocol buffer data, and protocol buffers have a
+[defined mapping](https://developers.google.com/protocol-buffers/docs/proto3#json)
+to JSON, this creates a natural mapping of CEL to JSON. This creates an exact
+bidirectional mapping between JSON types and a subset of CEL data:
+
+JSON Type | CEL Type
+--------- | ----------------------------------------------
+`null`    | `null`
+Boolean   | `bool`
+Number    | `double` (except infinities or NaN)
+String    | `string`
+Array     | `list` of elements in this table
+Object    | `map` (with string keys, values in this table)
+
+We define JSON mappings for much of the remainder of CEL data, but note that
+this data will not map back in to CEL as the same value:
+
+CEL Data                                             | JSON Data
+---------------------------------------------------- | ---------
+`int`                                                | Number if in interoperable range, otherwise decimal String.
+`uint`                                               | Number if in interoperable range, otherwise decimal String.
+double infinity                                      | String `"Infinity"` or `"-Infinity"`
+double NaN                                           | String "NaN"
+`bytes`                                              | String of base64-encoded bytes
+message                                              | JSON conversion of protobuf message.
+`list` with non-convertible elements                 | none
```

Comment:
Is it worth mentioning the lists and maps with convertible elements?

Reply:
Yes.  I've updated the list/map descriptions here and in the table above to distinguish bi-convertible, convertible, and non-convertible values.